src/Weeder/Main.hs

```diff
@@ -118,17 +134,19 @@ main = do
     hPutStrLn stderr $ "Did not find config: wrote default config to " ++ configPath
     writeFile configPath (configToToml defaultConfig)
 
-  (exitCode, _) <-
-    decodeConfig noDefaultFields configPath
-      >>= either throwIO pure
-      >>= mainWithConfig hieExt hieDirectories requireHsFiles
-
-  exitWith exitCode
+  decodeConfig noDefaultFields configPath
+    >>= either handleConfigError pure
+    >>= mainWithConfig hieExt hieDirectories requireHsFiles
   where
+    handleConfigError e = do
+      hPrint stderr e
+      exitWith (ExitFailure 3)
```

Comment:
I think it would be nice to have a table of these special exit codes. We could use either pattern synonyms (which I like because we can name them like constructors):

```haskell
pattern ExitConfigFailure = ExitFailure 3
```

or of course a regular top-level definition:

```haskell
exitConfigFailure :: ExitCode
exitConfigFailure = ExitFailure 3
```